app/src/main/java/org/oppia/android/app/player/state/ConfettiConfig.kt

```diff
@@ -0,0 +1,92 @@
+package org.oppia.android.app.player.state
+
+import android.content.Context
+import androidx.core.content.ContextCompat.getColor
+import nl.dionsegijn.konfetti.KonfettiView
+import nl.dionsegijn.konfetti.models.Shape
+import nl.dionsegijn.konfetti.models.Size
+import org.oppia.android.R
+
+/** A sealed class defining the standard configurations for various confetti animations */
+sealed class ConfettiConfig() {
+  internal abstract val context: Context
+  internal abstract val confettiView: KonfettiView
+  internal abstract val minSpeed: Float
+  internal abstract val maxSpeed: Float
+  internal abstract val sizeInDp: Size
+  // Confetti pieces with mass make the animation more active and dynamic.
+  internal abstract val sizeWithMass: Size
+  internal abstract val numPieces: Int
+  internal abstract val shapes: Array<Shape>
+
+  fun startConfettiBurst(
+    xPosition: Float,
+    yPosition: Float,
+    timeToLiveMs: Long,
+    delayMs: Long,
+    minAngle: Double,
+    maxAngle: Double,
+    colorsList: List<Int>
+  ) {
+    val colors = colorsList.map { getColor(context, it) }
+    confettiView.build()
+      .setDelay(delayMs)
+      .setFadeOutEnabled(true)
+      .addColors(colors)
+      .setDirection(minAngle, maxAngle)
+      .setSpeed(minSpeed, maxSpeed)
+      .setTimeToLive(timeToLiveMs)
+      .addShapes(*shapes)
+      .addSizes(sizeInDp, sizeWithMass)
+      .setPosition(xPosition, yPosition)
+      .burst(numPieces)
+  }
+
+  /** A configuration for large bursts of confetti. */
+  class LargeConfettiBurst(
+    override val context: Context,
+    override val confettiView: KonfettiView
```

Comment:
I think if you pass in the context & view to the show method, and also use arguments rather than values to initialize the ConfettiConfig class, you might be able to get away with making each of these child classes singleton objects, instead. It's a small win, but it's kind of nice to avoid needing to instantiate classes that need to be carefully managed (especially since they contain references to lifecycle-bound classes).

Reply:
Ooh as a Singleton class of each type of confetti, I see. I think the tradeoff is then the assembler must then pass the context and view around within itself to be able to initiate the animation so giving the configuration the view and context cleans up the assembler a bit. I kind of liked the how each confetti configuration could be tied to an individual instance of the animation so then, for example, two different confetti animations with the same configuration but in different areas / instances can be independently managed -- does that make sense?

Also, what do you mean by using arguments instead of values? Do you mean passing in the constants (size, shape, speed, etc.) when the configuration is constructed rather than standardizing the configuration with these constants? I think doing so this way would make the ConfettiConfig class somewhat redundant because the assembler would then have to instantiate all the configurations anyways.

Reply:
Done
Clarified offline, but ended up using enums instead of a sealed class so each configuration is just a single instance of a configuration.